linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -666,7 +666,7 @@ private[emitter] object CoreJSLib {
           if (implementedInObject) {
             val staticObjectCall: Tree = {
               val fun = classVar("c", ObjectClass).prototype DOT genName(methodName)
-              Return(Apply(fun DOT "call", instance :: args))
+              Return(Apply(genIdentBracketSelect(fun, "call"), instance :: args))
```

Comment:
That I wouldn't do. I'm worried that it could prevent GCC from recognizing that as a super call, hence preventing some optimizations. I think we should always preserve `.call` as true dot-selects.

Reply:
I'll try this out :)

Reply:
Seems indeed to be the case. I have also reverted the commit for `Array.length`. We can expect similar behavior (e.g. `[1,2,3].length` v.s. `[1,2,3]["length"]`)